include/io/io_mesh_ascii.tcc

```diff
@@ -691,6 +691,62 @@ std::vector<std::tuple<mpm::Index, unsigned, int, double>>
   return constraints;
 }
 
+//! Return cohesion constraints of particles
+template <unsigned Tdim>
+std::vector<std::tuple<mpm::Index, unsigned, int, double, double, int>>
+    mpm::IOMeshAscii<Tdim>::read_cohesion_constraints(
+        const std::string& cohesion_constraints_file) {
+
+  // Nodal cohesion constraints
+  std::vector<std::tuple<mpm::Index, unsigned, int, double, double, int>>
+      constraints;
+  constraints.clear();
+
+  // input file stream
+  std::fstream file;
+  file.open(cohesion_constraints_file.c_str(), std::ios::in);
+
+  try {
+    if (file.is_open() && file.good()) {
+      // Line
+      std::string line;
+      while (std::getline(file, line)) {
+        boost::algorithm::trim(line);
+        std::istringstream istream(line);
+        // ignore comment lines (# or !) or blank lines
+        if ((line.find('#') == std::string::npos) &&
+            (line.find('!') == std::string::npos) && (line != "")) {
+          // ID
+          mpm::Index id;
+          // Direction
+          unsigned dir;
+          // Sign
+          int sign;
+          // Cohesion
+          double cohesion;
+          // Cell height
+          double h_min;
```

Comment:
I am not sure I understand why we need to define `h_min` in the input for every node. In the current stage, we only use a constant h_min for all elements, is that possible to use the function `compute_average_cell_size()`?